src/jobs/jobs.controller.ts

```diff
@@ -368,11 +404,20 @@ export class JobsController {
             jobCreateDto.ownerUser,
           );
         }
-
         jobInstance.ownerUser = jobUser?.username as string;
         jobInstance.contactEmail = jobUser?.email as string;
         jobInstance.ownerGroup = jobCreateDto.ownerGroup;
```

Comment:
What is the expected result of these lines? I tested it by logging in as an admin and creating a job without providing ownerUser or ownerGroup. I am printing the `jobInstance` after these lines but everything seems to be undefined: createdBy, ownerGroup,, ownerUser, contactEmail are all undefined.

Reply:
"createdBy" field should be created by `utils.addCreatedByFields` from `jobs.service` and when I am logged in, it usually fills that field. Are you sure you were logged in? For the other fields, they are left empty for the case where admin creates a job for anonymous user. Then it's important to specify the email. For other cases, where users exist in the database, the email is going to be grabbed from the user data.